azurerm/resource_arm_backup_policy_file_share_test.go

```diff
@@ -0,0 +1,249 @@
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+		vaultName := rs.Primary.Attributes["recovery_vault_name"]
+		policyName := rs.Primary.Attributes["name"]
+
+		resp, err := client.Get(ctx, vaultName, resourceGroup, policyName)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+
+			return err
+		}
+
+		return fmt.Errorf("Recovery Services Vault Policy still exists:\n%#v", resp)
+	}
+
+	return nil
+}
+
+func testCheckAzureRMBackupProtectionPolicyFileShareExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := testAccProvider.Meta().(*ArmClient).RecoveryServices.ProtectionPoliciesClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %q", resourceName)
+		}
+
+		vaultName := rs.Primary.Attributes["recovery_vault_name"]
+		policyName := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Recovery Services Vault %q Policy: %q", vaultName, policyName)
+		}
+
+		resp, err := client.Get(ctx, vaultName, resourceGroup, policyName)
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Recovery Services Vault Policy %q (resource group: %q) was not found: %+v", policyName, resourceGroup, err)
+			}
+
+			return fmt.Errorf("Bad: Get on recoveryServicesVaultsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMBackupProtectionPolicyFileShare_base(rInt int, location string) string {
+	return fmt.Sprintf(` 
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_recovery_services_vault" "test" {
+  name                = "acctest-%[1]d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku                 = "Standard"
+}
+`, rInt, location, strconv.Itoa(rInt)[12:17])
+}
+
+func testAccAzureRMBackupProtectionPolicyFileShare_basicDaily(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_backup_policy_file_share" "test" {
+  name                = "acctest-%d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  recovery_vault_name = "${azurerm_recovery_services_vault.test.name}"
+
+  backup {
+    frequency = "Daily"
+    time      = "23:00"
+  }
+
+  retention_daily {
+    count = 10
+  }
+}
+`, testAccAzureRMBackupProtectionPolicyFileShare_base(rInt, location), rInt)
+}
+
+func testAccAzureRMBackupProtectionPolicyFileShare_updateDaily(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_backup_policy_file_share" "test" {
+  name                = "acctest-%d"
```

Comment:
Could we put some capitals in here to check the API handles them correctly?
```suggestion
  name                = "acctest-PFS-%d"
```